src/compute-client/src/protocol/response.rs

```diff
@@ -127,6 +127,21 @@ pub enum ComputeResponse<T = mz_repr::Timestamp> {
     /// [`CreateDataflow` command]: super::command::ComputeCommand::CreateDataflow
     /// [`AllowCompaction` command]: super::command::ComputeCommand::AllowCompaction
     SubscribeResponse(GlobalId, SubscribeResponse<T>),
+    /// `CopyToResponse` reports the results emitted by an active COPY ... TO command.
+    ///
+    /// The replica must send exactly one `CopyToResponse` for a previously run
+    /// [`CreateDataflow` command].
+    ///
+    /// Once a copy is done,
+    ///
+    ///   * It must no longer read from its inputs.
+    ///   * The replica must not send further `CopyToResponse`s for that command.
+    ///
+    /// The replica must not send `CopyToResponse`s for copy tos that have not been
+    /// created previously by a [`CreateDataflow` command].
+    ///
+    /// [`CreateDataflow` command]: super::command::ComputeCommand::CreateDataflow
+    CopyToResponse(GlobalId, CopyToResponse),
```

Comment:
From our chat today, renaming the sink to `CopyToS3Oneshot` from `S3Oneshot`. The `CopyToResponse` buffer and all the plumbing should be re-usable if in case we add more compute sinks say like GCP or Azure's blob storage.

Reply:
I will actually do the rename in a separate PR to reduce the diffs in this PR.